### PR TITLE
Project: Private Events

### PR DESCRIPTION
**Pull request by mariordgez**
The purpose of this particular project in a general way was to use Active record associations to join different models, as well as learning how to use queries in a more useful way, this project is able to accomplish these functions:
- Create and log in as a new user
- Once logged in, you can create events
- The user is able to 'attend' events created by other users
- The index and the User's show page have their listed events sorted by past and future events using two different methods (view logic and scopes)
- The event show page allows you to attend the event as well as listing the attendees for the event

This is generally speaking how this project works, I hope you like it, have a wonderful day.

### CODE REVIEW

File: app/controllers/attendees_controller.rb [hunk 1]
```diff
@@ -0,0 +1,39 @@
+class AttendeesController < ApplicationController
+  def new
+    helpers.signed_in_only
+    # @event = Event.new
```
Comment (SajjadAhmad14) [inline app/controllers/attendees_controller.rb]: Please remove commented code. It makes the whole code base unclean.

File: app/controllers/events_controller.rb [hunk 2]
```diff
@@ -1,31 +1,32 @@
...
   # GET /events/1 or /events/1.json
-  def show
-  end
+  def show; end
 
   # GET /events/new
   def new
-    @event = Event.new
+    helpers.signed_in_only
+    # @event = Event.new
```
Comment (SajjadAhmad14) [inline app/controllers/events_controller.rb]: Please remove commented code. It makes the whole code base unclean.

**Review by SajjadAhmad14 — CHANGES_REQUESTED** [review]
# Changes Required :recycle:
> "Change is inevitable. Change is constant. -Benjamin Disraeli"
## Hello, you have been doing a tremendous job so far :ok_hand:
## To Highlight 🎉
- [x] Git flow is used properly.
- [x] README has all the steps to run the project locally.
- [x] Project runs perfectly.
## To Improve 🙏
- [x] Please add `yarn install` or `npm install` so that packages can be installed related to browser functionality.
- [x] Please remove commented code from controllers and all other files.
- [x] #index action of events controller shoul have @past_events = Event.previous_events. Please see this [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/past_upcoming_events_requirement.png).
- [x] #index action of events controller shoul have @upcoming_events = Event.upcoming_events. Please see this [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/past_upcoming_events_requirement.png).
- [x] #show action of users controller should have @created_events = current_user.events. Please see this [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/current_user_show_requirement.png).
- [x] Models should have unit tests for validations and associations using Rspec. Please see this [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/unit_tests_requirement.png).
- [x] Authentication should have integration tests using Rspec + Capybara. Please see this [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/integration_tests_requirement.png).
- [x] Event management features have integration tests using Rspec + Capybara. Please see this [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/integration_tests_requirement.png).
# Optional
- [x] Views should be completely separate from app logic. You can move all view-related logic to app/helpers.
### Happy Coding ☺️
## If you have any questions you can comment here in this PR or can always connect with me on slack ( @Sajjad Ahmad ).

**Review by voscarmv — APPROVED** [review]
# Project status: Approved 💯 ✨ 😎 👍 
### You may merge this branch and move forward ⛙🛣️ 

### Questions or comments? Reach me on [Twitter](https://twitter.com/voscarmv) or [LinkedIn](https://linkedin.com/in/oscar-mier) 🧑‍💻🗨️🧑‍💻
# Happy Coding! 🧠 ⌨️ 🖱️ 🖥️

**Pull request merged**